### DISCUSSION TITLE
Business email validation

### DISCUSSION THREAD

**Discussion by ganatantra-kumar** [elementor/elementor#18398] [Developers Community]
With the Below code, I am trying to select the Label Email and make validation. But don't know why this code is not being triggered. Can someone please check and advise.

Note: this is being used in WordPress with Elementor Builder.

The snippet is attached:

[Business email validation for Elementor(Wordpress).txt](https://github.com/elementor/elementor/files/8458476/Business.email.validation.for.Elementor.Wordpress.txt)
<img width="694" alt="Business email validation for Elementor(Wordpress)" src="https://user-images.githubusercontent.com/26713493/162607212-9713b26e-6049-4736-88b3-933a569547b6.png">

**Comment by ganatantra-kumar** [comment]
`<html>
<body>


<script>
--
  | var label = 'Email';
  | var standardEmail = /^(?!.+@(gmail\|google\|yahoo\|outlook\|hotmail\|msn)\..+)(.+@.+\..+)$/;
  | var errorMessage = 'field: Please enter your business email address.';
  |  
  | document.querySelector("input[name='" + label + "']").addEventListener("input", function(e){
  | e.target.value = e.target.value.toLowerCase();
  | });
  |  
  | window.__custom_form_validations = [
  | {
  | fieldName: label,
  | validationFn: function (input) {
  | return {
  | isValid: standardEmail.test(input.value),
  | message: window._Translate.get(errorMessage),
  | };
  | },
  | },
  | ];
  | </script>


</body>
</html>`

**Reply by thinktrans** [reply]
While we are at it, does elementor support internationalized e-mail addresses? 

They are new (not so new from technical implementation POV but more from the wider community acceptance POV) entrants on the block and the prominent software platforms need to be aware of their requirements.

Happy to add more to the discussion if the community finds this interesting enough to work upon.

**Comment by 0jscsshtml** [comment]
```
function blacklisted_email_domain( $record, $ajax_handler ) { 
	$form_name = $record->get_form_settings('form_name');

	if ('YOUR_FORM_NAME' !== $form_name) {
		return;
	}

	$email_field = $record->get_field( [
		'id' => 'YOUR_EMAIL_FIELD_ID'
	] );

	$black_list_domains = [
		'gmail.com',
		'yahoo.com',
		'outlook.com',
		'hotmail.com',
		'msn.com',
	];
	
	$email_domain = explode( '@', $email_field['YOUR_EMAIL_FIELD_ID']['value'] )[1];
	
	if ( in_array( $email_domain, $black_list_domains ) ) {
		$ajax_handler->add_error( $email_field['YOUR_EMAIL_FIELD_ID']['id'], 'Invalid Email address, emails from the domain ' . $email_domain . ' is not allowed. Please use your business email address.' );
		return;
	}

}
add_action( 'elementor_pro/forms/validation', 'blacklisted_email_domain', 10, 2); 
```
